node.c

```diff
@@ -0,0 +1,84 @@
+#include "node.h"
+#include "board.h"
+
+
+void initializeRoot(node* root, struct board *boardState){
+
+    // Setting default values
+    root->parent = NULL;
+    root->childMoves = (int *)malloc(4 * sizeof(int));
+    root->numChildren = 0;
+    root->numWins = 0;
+    copyBoard(&boardState, &root->board);
+    root->numSimulations = 0;
+    root->children = (node **)malloc(2 * sizeof(node*));
+}
+
+
+node* expand(node* parentNode, int* action){
+
+    // Dynamically allocate new struct node
+    node* newNode = (node *)malloc(sizeof(node));
+
+    // Setting member variables for new node
+    newNode->parent = parentNode;
+    newNode->childMoves = (int *)malloc(4 * sizeof(int));
+    newNode->numChildren = 0;
+    newNode->numWins = 0;
+    newNode->numSimulations = 0;
+    newNode->children = (node **)malloc(2 * sizeof(node*));
+    printf("Mallocing space for new node to have 2 children\n");
+
+    copyBoard(&parentNode->board, &newNode->board);
+    placePiece(&newNode->board, action);
+
+
+    // Update parent node
+    if ((parentNode->numChildren != 0) && (parentNode->numChildren % 2 == 0)){
```

Comment:
You should only reallocate when the number of children is a power of 2. This can be tested via the equation `x & (x-1) == 0` which I picked up from [this StackOverflow post](http://stackoverflow.com/questions/600293/how-to-check-if-a-number-is-a-power-of-2). The first half of this if statement is fine though.

My bad, I was the one who suggested you do it by checking if the number of children is divisible by two.